core/app/models/spree/order.rb

```diff
@@ -524,9 +525,15 @@ def ensure_shipping_address
     end
 
     def create_proposed_shipments
-      adjustments.shipping.destroy_all
-      shipments.destroy_all
-      self.shipments = Spree::Stock::Coordinator.new(self).shipments
+      if completed?
+        raise CannotRebuildShipments.new(Spree.t(:cannot_rebuild_shipments_order_completed))
+      elsif shipments.any? { |s| !s.pending? }
```

Comment:
Just curious what situation this is intended for? I don't think shipments can leave pending until the order is completed?